city_scrapers/spiders/chi_ssa_18.py

```diff
@@ -0,0 +1,117 @@
+import re
+from datetime import datetime
+
+from city_scrapers_core.constants import COMMISSION
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+
+
+class ChiSsa18Spider(CityScrapersSpider):
+    name = "chi_ssa_18"
+    agency = "Chicago Special Service Area #18 North Halsted"
+    timezone = "America/Chicago"
+    start_urls = ["https://northalsted.com/community/"]
+    location = {
+        'name': 'Center on Halsted',
+        'address': '3656 N Halsted St, Chicago IL 60613, Conference Room 200 on the 2nd Floor'
+    }
+    meeting_year = ''  # This is used to track year in header elements & append to meetings
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` Meeting items.
+
+        Change the `_parse_title`, `_parse_start`, etc methods to fit your scraping
+        needs.
+        """
+        table = response.xpath('//*[@id="page_content_wrapper"]/div/div/div/div[1]/div[3]')
+
+        self._validate_location(table)
+        self._validate_meeting_times(table)
+
+        for item in table.xpath('.//h3 | .//p'):
+            if 'h3' in item.get():
+                ChiSsa18Spider.meeting_year = item.xpath('descendant-or-self::text()')\
+                    .re_first(r'\d\d\d\d')
+                continue
+            if '<p>' in item.get():
+                split_test = item.get().split("<br>")
+                for split_string in split_test:
+                    meeting_date_match = re.search(r'([A-Z]\w{2,}\s\d\d?)', split_string)
+                    if meeting_date_match and ChiSsa18Spider.meeting_year:
+                        try:
+                            converted_date = \
+                                self._convert_date(
+                                    meeting_date_match.group(), ChiSsa18Spider.meeting_year
+                                    )
+                        except ValueError:
+                            continue
```

Comment:
If `converted_date` is required to keep iterating then instead of catching `ValueError` the `else` condition in here can be `continue`

Reply:
I didn't make changes for this. I think I like the control flow better here since the 'else' would have to come after the creation of the meeting object. Here, converted date is required to create a meeting in the first place. Also, there are some h3 headers before the actual meeting items that actually return a string in response to the regex, so the 'try' here ensures that any value errors from passing the string doesn't stop the spider.

Reply:
Gotcha, I see what you're saying. I wasn't clear on this, but generally we try to exit the loop as soon as possible rather than nesting more because it's easy to read, so it wouldn't necessarily be the `else` condition, but more something like this:

```python
if not (meeting_date_match and meeting_year):
  continue

meeting = Meeting(...
```
This way, it's clear what we're looking for and the nesting is a little less deep. As for the errors, typically we want the scrapers to fail rather than behave in a way we aren't expecting. Otherwise we sometimes end up with bad data that isn't as easy to track as getting an error message. Let me know what you think, thanks!

Reply:
This makes sense and the code is much easier to understand! I updated the code as you described and everything works out!